src/main/java/core/basesyntax/Bulldozer.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Bulldozer extends Machine {
+    private boolean hasCaterpillar;
+
+    Bulldozer(String model, String fuel, double engineCapacity, boolean hasCaterpillar) {
+        super(model, fuel, engineCapacity);
+        this.hasCaterpillar = hasCaterpillar;
+    }
+
+    public boolean isHasCaterpillar() {
+        return hasCaterpillar;
+    }
+
+    public void setCaterpillar(boolean hasCaterpillar) {
+        hasCaterpillar = hasCaterpillar;
+    }
+
+    @Override
+    public void doWork() {
+        System.out.println("Bulldozer is started its work");
+    }
+
+    @Override
+    public void stopWork() {
+        System.out.println("Bulldozer is stopped its work");
+    }

```

Comment:
Please pay close attention to the task description, no need to declare all these fields, the task is just to implement doWork and stopWork in each child class, that's all. This comment applies to all the other classes